tests/test_validate_parameters.py

```diff
@@ -10,14 +10,60 @@
 
 import pytest
 import yaml
+from click.testing import CliRunner
 
+from reana_client.utils import cwl_load
 from reana_client.validation import (
     _validate_dangerous_operations,
     _validate_serial_parameters,
     _validate_yadage_parameters,
+    _validate_cwl_parameters,
 )
 
 
+def test_validate_parameters_cwl(
+    create_cwl_yaml_workflow_schema,
+    cwl_workflow_spec_step,
+    cwl_workflow_spec_correct_input_param,
+    cwl_workflow_spec_wrong_input_param,
+    capsys,
+):
+    """Validate parameters for CWL workflows."""
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        with open("main.cwl", "w") as f:
+            f.write(cwl_workflow_spec_step)
+        with open("test.tool", "w") as g:
+            g.write(cwl_workflow_spec_correct_input_param)
+        reana_yaml = yaml.load(create_cwl_yaml_workflow_schema, Loader=yaml.FullLoader)
+        reana_yaml["workflow"]["specification"] = cwl_load(
+            reana_yaml["workflow"].get("file")
+        )
```

Comment:
This is kind of repeated between the tests, maybe we can wrap it into a function inside this test that gets the workflow spec as param?